homeassistant/components/growatt_server/sensor.py

```diff
@@ -0,0 +1,161 @@
+"""Read status of growatt inverters."""
+import datetime
+import logging
+import json
+
+import growattServer
+
+import voluptuous as vol
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_USERNAME, CONF_PASSWORD
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PLANT_ID = "plant_id"
+DEFAULT_PLANT_ID = "0"
+DEFAULT_NAME = "Growatt"
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PLANT_ID, default=DEFAULT_PLANT_ID): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Growatt sensor."""
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+    plant_id = config.get(CONF_PLANT_ID)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.